lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -1108,6 +1111,23 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
       // Make sure that if operand is generic that its primary archetypes match
       // the function context.
       checkLegalType(I->getFunction(), operand.get(), I);
+
+      // If we are not in OSSA, our operand constraint should be none if we have
+      // a type dependent operand and OwnershipKind::Any,
+      // UseLifetimeConstraint::NonLifetimeEnding.
+      if (!I->getFunction()->hasOwnership()) {
+        auto constraint = operand.getOwnershipConstraint();
+        require(operand.isTypeDependent() || constraint.hasValue(),
```

Comment:
`require(operand.isTypeDependent() || constraint.hasValue()`
you can move this inside the `if (!operand.isTypeDependent()) ` check